### DISCUSSION TITLE
Mandatory 2FA: Meaning of "contribute code"

### DISCUSSION THREAD

**Discussion by Journeyman-Joe** [community/community#57366] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

The GitHub blog article containing the schedule for mandatory 2FA states that the requirement applies to "_all developers who contribute code on GitHub.com_".

[Raising the bar for software security: GitHub 2FA begins March 13](https://github.blog/2023-03-09-raising-the-bar-for-software-security-github-2fa-begins-march-13/)

Is an account holder who pushes code **only** to his / her own repositories (i.e., is not making pull requests to other projects, or repositories owned by other accounts) considered to be "_contributing code_"?  Will such accounts be required to enable 2FA before the end of 2023?

**Comment by ImaniAN** [comment]
The 2FA applies to all GitHub accounts and users regardless of where or how you contribute. 

if you read the article that you sent, you can find the detail of how this will done. 

GitHub will start with organizations and small teams by selecting a few accounts to do the 2FA... 

As time goes other accounts will be asked to do the 2FA ... The article clearly describes a plan of action to execute the enrollment process

**Comment by airtower-luna** [comment]
This older article linked right at the top of the one you linked is a bit more clear: https://github.blog/2022-05-04-software-security-starts-with-the-developer-securing-developer-accounts-with-2fa/

> While we strongly believe 2FA for active contributors (for example, those who commit code, open or merge pull requests, use Actions, or publish packages) is the right thing to do, we also want to ensure a smooth and accessible experience, […]

**Reply by ImaniAN** [reply]
yes, this clearly states that "GitHub will require all users who contribute code on GitHub.com to enable one or more forms of two-factor authentication (2FA) by the end of 2023."

Hopefully, that part answers the question that was initially asked
